pkg/ami/api.go

```diff
@@ -60,8 +66,9 @@ func IsAvailable(api ec2iface.EC2API, id string) (bool, error) {
 // FindImage will get the AMI to use for the EKS nodes by querying AWS EC2 API.
 // It will only look for images with a status of available and it will pick the
 // image with the newest creation date.
-func FindImage(api ec2iface.EC2API, namePattern string) (string, error) {
+func FindImage(api ec2iface.EC2API, namePattern string, imageFamily string) (string, error) {
 	input := &ec2.DescribeImagesInput{
+		Owners: []*string{aws.String(ImageFamilyToAccountID[imageFamily])},
```

Comment:
This a little too optimistic, we should check if the key exists, so we don't panic if it doesn't.

Reply:
Please move the look-up  into the caller, and the map declaration should live in the same file also:

https://github.com/weaveworks/eksctl/blob/3d20b573fcc5b495ba6729ba4affdc25a285639a/pkg/ami/auto_resolver.go#L50